apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOverview/WeaponSwapModal.tsx

```diff
@@ -46,6 +57,17 @@ export default function WeaponSwapModal({ onChangeId, weaponTypeKey, show, onClo
 
   return <ModalWrapper open={show} onClose={onClose} >
     <CardDark>
+      <Suspense fallback={false}>
+        <WeaponSelectionModal show={newWeaponModalShow} onHide={() => setnewWeaponModalShow(false)} onSelect={newWeapon} />

```

Comment:
Can you add a filter for the character's weapon type? `filter={(sheet) => sheet.weaponType === char.weaponType`